### PR TITLE
feat(react-await): add `awaitOptions`

### PR DESCRIPTION
**Pull request by SEOKKAMONI**
# Overview
resolved https://github.com/toss/suspensive/issues/433

- awaitOptions (experimental)
- add tests for awaitOptions, useAwait

## PR Checklist

- [x] I did below actions if need

1. I read the [Contributing Guide](https://github.com/toss/suspensive/blob/main/CONTRIBUTING.md)
2. I added documents and tests.

### CODE REVIEW

File: docs/suspensive.org/src/pages/docs/react-await/awaitOptions.en.mdx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+# awaitOptions
+
+<Callout type='experimental'>
+
+`<Await/>` is experimental feature, this interfaces could be changed
+
+</Callout>
+
+`awaitOptions` combines a key and a function to help manage the options in the current `<Await />` component.
```
Comment (SEOKKAMONI) [inline docs/suspensive.org/src/pages/docs/react-await/awaitOptions.en.mdx:11]: If the interface changes in the future, we will also change the document. But I thought it was right to specify the <Await /> component to deliver explicitly as of now

File: packages/react-await/src/index.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
 export { Await, useAwait, awaitClient } from './Await'
+export { awaitOptions } from './awaitOptions'
 
-export type { AwaitProps } from './Await'
+export type { AwaitProps, AwaitOptions } from './Await'
```
Comment (SEOKKAMONI) [inline packages/react-await/src/index.ts:4]: With the development of 'awaitOptions', I thought that the AwaitOptions type should also be given access, so I allowed it as follows.

File: packages/react-await/src/awaitOpetions.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    render(
+      <Suspense fallback={FALLBACK}>
+        <Await options={options}>{({ data }) => <>{data}</>}</Await>
+      </Suspense>
+    )
+
+    expect(await screen.findByText(TEXT)).toBeInTheDocument()
+  })
+
+  it('should ve used with useAwait', async () => {
```
Comment (SEOKKAMONI) [inline packages/react-await/src/awaitOpetions.spec.tsx:22]:
```suggestion
  it('should be used with useAwait', async () => {
```

File: packages/react-await/src/Await.tsx [hunk 4]
```diff
@@ -4,7 +4,7 @@ import { hashKey } from './utils'
 
 export type Key = Tuple
 
-type AwaitOptions<TData, TKey extends Key> = {
+export type AwaitOptions<TData, TKey extends Key> = {
```
Comment (manudeli) [inline packages/react-await/src/Await.tsx:7]:
```suggestion
/**
 * @experimental This is experimental feature.
 */
export type AwaitOptions<TData, TKey extends Key> = {
```

**Review by manudeli — APPROVED** [review]
🚀🚀🚀

**Pull request merged**